### ISSUE TITLE
Not happy with kickstart script

### ISSUE THREAD

**Issue by cryptofuture** [firehol/netdata#2342]
It miss `--dont-start-it` from netdata-installer.sh and it downloads netdata source to `/usr/src/netdata` while I prefer `/opt/netdata` and old installation way respects source directory you run script from. More above I found out that non interactive installs cron job by default, and I do it manually. Also my-netdata.io just drops connections to several ips, or ISP blocks it.
Basically, I can restore [part of old install instructions](https://github.com/firehol/netdata/wiki/Installation/c2f0d2190e00ceae6564789c701b7f7f2db0eb76) as addition to current one from the old revision.
For example the way I do it now:
```
cd /opt/
git clone https://github.com/firehol/netdata.git --depth=1
cd netdata
curl -Ss 'https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh' >/tmp/kickstart.sh && bash /tmp/kickstart.sh -i netdata (actually I install packages manually)
./netdata-installer.sh --dont-wait --dont-start-it
```
However I think better solution will be just to add `--dont-start-it` and `--src-dir=/opt/netdata` `--no-cron` to kickstarter script. Also it may be viable to add `--no-systemd`.
Ideas?

**Comment by ktsaou** [comment]
nice suggestions...

**Comment by ktsaou** [comment]
ok, in PR #2354 

1. added option `--src-dir PATH` 
2. added option `--no-updates`
3. option `--dont-start-it` was already supported (all installer options can be appended)

I think `--no-systemd` is not needed (why to do this?)

**Comment by cryptofuture** [comment]
Why `if [ ! -d "${SOURCE_DST}/netdata.git" ]` but not just `netdata` is it really more viable? I think its pretty obvious source updates from git, but directory name with .git in name looks really alien.

**Comment by ktsaou** [comment]
Answered on the PR.

**Comment by ktsaou** [comment]
merged it.

**Comment by cryptofuture** [comment]
I still have problem with a kickstart script (actually with install-required-packages.sh) with `--dont-wait`. Not sure it worth separate issue.
Salt (init.sls)
```
install_git:
  pkg.installed:
    - pkgs:
      - git

netdata_install:
  cmd.run:
        - name: |
            cd /tmp && git clone https://github.com/ktsaou/netdata
            /tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait
```

Result:
```
----------
          ID: install_git
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 13:42:16.091510
    Duration: 967.632 ms
     Changes:   
----------
          ID: netdata_install
    Function: cmd.run
        Name: cd /tmp && git clone https://github.com/ktsaou/netdata
/tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait

      Result: True
     Comment: Command "cd /tmp && git clone https://github.com/ktsaou/netdata
              /tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait
              " run
     Started: 13:42:17.069793
    Duration: 10384.198 ms
     Changes:   
              ----------
              pid:
                  17461
              retcode:
                  0
              stderr:
                  Cloning into 'netdata'...
                  /tmp/netdata/kickstart.sh: 220: [: Illegal number: 
                   --- Downloading script to detect required packages... --- 
                  [/tmp]$ /usr/bin/wget -O - https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh
                  --2017-06-18 13:42:21--  https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh
                  Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.120.133
                  Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.120.133|:443... connected.
                  HTTP request sent, awaiting response... 200 OK
                  Length: 40336 (39K) [text/plain]
                  Saving to: 'STDOUT'
                  
                       0K .......... .......... .......... .........            100% 24.8M=0.002s
                  
                  2017-06-18 13:42:21 (24.8 MB/s) - written to stdout [40336/40336]
                  
                   OK   
                  
                   --- Running downloaded script to detect required packages... --- 
                  [/tmp]$ /bin/bash /tmp/netdata-kickstart-W3jZXX netdata
                  Loading /etc/os-release ...
                   > Checking if package 'autoconf' is installed...
                   > Checking if package 'autoconf-archive' is installed...
                  dpkg-query: no packages found matching autoconf-archive
                   > Checking if package 'autogen' is installed...
                  dpkg-query: no packages found matching autogen
                   > Checking if package 'automake' is installed...
                   > Checking if package 'pkg-config' is installed...
                  dpkg-query: no packages found matching pkg-config
                   > Checking if package 'curl' is installed...
                   > Checking if package 'zlib1g-dev' is installed...
                  dpkg-query: no packages found matching zlib1g-dev
                   > Checking if package 'uuid-dev' is installed...
                  dpkg-query: no packages found matching uuid-dev
                   > Checking if package 'libmnl-dev' is installed...
                  dpkg-query: no packages found matching libmnl-dev
                   > Checking if package 'python-yaml' is installed...
                  
                  The following command will be run:
                  
                   >> IMPORTANT << 
                      Please make sure your system is up to date
                      by running:   apt-get update  
                  
                  apt-get install autoconf autoconf-archive autogen automake curl libmnl-dev pkg-config uuid-dev zlib1g-dev 
                  
                  
                  apt-get install autoconf autoconf-archive autogen automake curl libmnl-dev pkg-config uuid-dev zlib1g-dev 
                   FAILED   
                  
                  It failed to install all the required packages, but I can try to install netdata.
                  /tmp/netdata/kickstart.sh: 298: read: arg count
                   --- OK, let's give it a try... --- 
                   --- Downloading netdata source code... --- 
                  [/tmp]$ /usr/bin/git clone https://github.com/firehol/netdata.git /opt/netdata.git
                  Cloning into '/opt/netdata.git'...
                   OK   
                  
                   --- Installing netdata... ---
              stdout:
                  System            : Linux
                  Operating System  : GNU/Linux
                  Machine           : x86_64
                  BASH major version: 4
                  
                  /etc/os-release information:
                  NAME            : Ubuntu
                  VERSION         : 16.04.2 LTS (Xenial Xerus)
                  ID              : ubuntu
                  ID_LIKE         : debian
                  VERSION_ID      : 16.04
                  
                  We detected these:
                  Distribution    : ubuntu
                  Version         : 16.04
                  Codename        : 16.04.2 LTS (Xenial Xerus)
                  Package Manager : install_apt_get
                  Packages Tree   : debian
                  Detection Method: /etc/os-release
                  Default Python v: 2 
                  
                  Reading package lists...
                  Building dependency tree...
                  Reading state information...
                  The following additional packages will be installed:
                    autotools-dev dpkg-dev guile-2.0-libs libc-dev-bin libc6-dev libdpkg-perl
                    libgc1c2 libltdl7 libopts25 libopts25-dev libsigsegv2 libunistring0
                    linux-libc-dev m4
                  Suggested packages:
                    gnu-standards autoconf-doc libtool gettext debian-keyring glibc-doc
                    manpages-dev
                  Recommended packages:
                    autogen-doc build-essential fakeroot libalgorithm-merge-perl manpages-dev
                    libfile-fcntllock-perl
                  The following NEW packages will be installed:
                    autoconf autoconf-archive autogen automake autotools-dev curl dpkg-dev
                    guile-2.0-libs libc-dev-bin libc6-dev libdpkg-perl libgc1c2 libltdl7
                    libmnl-dev libopts25 libopts25-dev libsigsegv2 libunistring0 linux-libc-dev
                    m4 pkg-config uuid-dev zlib1g-dev
                  0 upgraded, 23 newly installed, 0 to remove and 0 not upgraded.
                  Need to get 9214 kB of archives.
                  After this operation, 47.7 MB of additional disk space will be used.
                  Do you want to continue? [Y/n] Abort.
                  
                  
                  
                  We are very sorry!
                  
                  Installation of required packages failed.
                  
                  What to do now:
                  
                    1. Make sure your system is updated.
                       Most of the times, updating your system will resolve the issue.
                  
                    2. If the error message is about a specific package, try removing
                       that package from the command and run it again.
                       Depending on the broken package, you may be able to continue.
                  
                    3. Let us know. We may be able to help.
                       Open a github issue with the above log, at:
                  
                             https://github.com/firehol/netdata/issues
                  
                  
                  ./netdata-installer.sh --dont-start-it --dont-wait
```
It looks like it still waits for `Y` answer
I used ktsaou/netdata kickstart script here, since it already was with changes before, you merge it.

**Comment by cryptofuture** [comment]
I still have problem with a kickstart script (actually with install-required-packages.sh) with `--dont-wait`. Not sure it worth separate issue.
Salt (init.sls)
```
install_git:
  pkg.installed:
    - pkgs:
      - git

netdata_install:
  cmd.run:
        - name: |
            cd /tmp && git clone https://github.com/ktsaou/netdata
            /tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait
```

Result:
```
----------
          ID: install_git
    Function: pkg.installed
      Result: True
     Comment: All specified packages are already installed
     Started: 13:42:16.091510
    Duration: 967.632 ms
     Changes:   
----------
          ID: netdata_install
    Function: cmd.run
        Name: cd /tmp && git clone https://github.com/ktsaou/netdata
/tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait

      Result: True
     Comment: Command "cd /tmp && git clone https://github.com/ktsaou/netdata
              /tmp/netdata/kickstart.sh --src-dir /opt --no-updates --dont-start-it --dont-wait
              " run
     Started: 13:42:17.069793
    Duration: 10384.198 ms
     Changes:   
              ----------
              pid:
                  17461
              retcode:
                  0
              stderr:
                  Cloning into 'netdata'...
                  /tmp/netdata/kickstart.sh: 220: [: Illegal number: 
                   --- Downloading script to detect required packages... --- 
                  [/tmp]$ /usr/bin/wget -O - https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh
                  --2017-06-18 13:42:21--  https://raw.githubusercontent.com/firehol/netdata-demo-site/master/install-required-packages.sh
                  Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.120.133
                  Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.120.133|:443... connected.
                  HTTP request sent, awaiting response... 200 OK
                  Length: 40336 (39K) [text/plain]
                  Saving to: 'STDOUT'
                  
                       0K .......... .......... .......... .........            100% 24.8M=0.002s
                  
                  2017-06-18 13:42:21 (24.8 MB/s) - written to stdout [40336/40336]
                  
                   OK   
                  
                   --- Running downloaded script to detect required packages... --- 
                  [/tmp]$ /bin/bash /tmp/netdata-kickstart-W3jZXX netdata
                  Loading /etc/os-release ...
                   > Checking if package 'autoconf' is installed...
                   > Checking if package 'autoconf-archive' is installed...
                  dpkg-query: no packages found matching autoconf-archive
                   > Checking if package 'autogen' is installed...
                  dpkg-query: no packages found matching autogen
                   > Checking if package 'automake' is installed...
                   > Checking if package 'pkg-config' is installed...
                  dpkg-query: no packages found matching pkg-config
                   > Checking if package 'curl' is installed...
                   > Checking if package 'zlib1g-dev' is installed...
                  dpkg-query: no packages found matching zlib1g-dev
                   > Checking if package 'uuid-dev' is installed...
                  dpkg-query: no packages found matching uuid-dev
                   > Checking if package 'libmnl-dev' is installed...
                  dpkg-query: no packages found matching libmnl-dev
                   > Checking if package 'python-yaml' is installed...
                  
                  The following command will be run:
                  
                   >> IMPORTANT << 
                      Please make sure your system is up to date
                      by running:   apt-get update  
                  
                  apt-get install autoconf autoconf-archive autogen automake curl libmnl-dev pkg-config uuid-dev zlib1g-dev 
                  
                  
                  apt-get install autoconf autoconf-archive autogen automake curl libmnl-dev pkg-config uuid-dev zlib1g-dev 
                   FAILED   
                  
                  It failed to install all the required packages, but I can try to install netdata.
                  /tmp/netdata/kickstart.sh: 298: read: arg count
                   --- OK, let's give it a try... --- 
                   --- Downloading netdata source code... --- 
                  [/tmp]$ /usr/bin/git clone https://github.com/firehol/netdata.git /opt/netdata.git
                  Cloning into '/opt/netdata.git'...
                   OK   
                  
                   --- Installing netdata... ---
              stdout:
                  System            : Linux
                  Operating System  : GNU/Linux
                  Machine           : x86_64
                  BASH major version: 4
                  
                  /etc/os-release information:
                  NAME            : Ubuntu
                  VERSION         : 16.04.2 LTS (Xenial Xerus)
                  ID              : ubuntu
                  ID_LIKE         : debian
                  VERSION_ID      : 16.04
                  
                  We detected these:
                  Distribution    : ubuntu
                  Version         : 16.04
                  Codename        : 16.04.2 LTS (Xenial Xerus)
                  Package Manager : install_apt_get
                  Packages Tree   : debian
                  Detection Method: /etc/os-release
                  Default Python v: 2 
                  
                  Reading package lists...
                  Building dependency tree...
                  Reading state information...
                  The following additional packages will be installed:
                    autotools-dev dpkg-dev guile-2.0-libs libc-dev-bin libc6-dev libdpkg-perl
                    libgc1c2 libltdl7 libopts25 libopts25-dev libsigsegv2 libunistring0
                    linux-libc-dev m4
                  Suggested packages:
                    gnu-standards autoconf-doc libtool gettext debian-keyring glibc-doc
                    manpages-dev
                  Recommended packages:
                    autogen-doc build-essential fakeroot libalgorithm-merge-perl manpages-dev
                    libfile-fcntllock-perl
                  The following NEW packages will be installed:
                    autoconf autoconf-archive autogen automake autotools-dev curl dpkg-dev
                    guile-2.0-libs libc-dev-bin libc6-dev libdpkg-perl libgc1c2 libltdl7
                    libmnl-dev libopts25 libopts25-dev libsigsegv2 libunistring0 linux-libc-dev
                    m4 pkg-config uuid-dev zlib1g-dev
                  0 upgraded, 23 newly installed, 0 to remove and 0 not upgraded.
                  Need to get 9214 kB of archives.
                  After this operation, 47.7 MB of additional disk space will be used.
                  Do you want to continue? [Y/n] Abort.
                  
                  
                  
                  We are very sorry!
                  
                  Installation of required packages failed.
                  
                  What to do now:
                  
                    1. Make sure your system is updated.
                       Most of the times, updating your system will resolve the issue.
                  
                    2. If the error message is about a specific package, try removing
                       that package from the command and run it again.
                       Depending on the broken package, you may be able to continue.
                  
                    3. Let us know. We may be able to help.
                       Open a github issue with the above log, at:
                  
                             https://github.com/firehol/netdata/issues
                  
                  
                  ./netdata-installer.sh --dont-start-it --dont-wait
```
It looks like it still waits for `Y` answer
I used ktsaou/netdata kickstart script here, since it already was with changes before, you merge it.

**Comment by ktsaou** [comment]
You need to append the `netdata-installer.sh` options to `kickstart.sh`. The option `--dont-start-it` is a netdata-installer.sh option that kickstart.sh does not know, so it stops processing arguments and does not see `--dont-wait`.

I also saw the error about `UID` being empty. I am submitting a PR now.

**Comment by ktsaou** [comment]
fixed that one too PR #2358

**Comment by cryptofuture** [comment]
Sorry, was stupid `kickstart.sh --src-dir /opt --no-updates --dont-wait --dont-start-it` works fine.
